lib/pages/chat/chat_input_row.dart

```diff
@@ -27,53 +29,161 @@ class ChatInputRow extends StatelessWidget {
-                      SizedBox(
-                        height: ChatInputRowStyle.chatInputRowHeight,
-                        child: TwakeIconButton(
-                          size: ChatInputRowStyle.chatInputRowMoreBtnSize,
-                          tooltip: L10n.of(context)!.more,
-                          icon: Icons.add_circle_outline,
-                          onTap: () => controller.onSendFileClick(context),
+                    child: SocialMediaRecorder(
+                      radius: BorderRadius.circular(24),
+                      soundRecorderWhenLockedDecoration: BoxDecoration(
+                        borderRadius:
+                            ChatInputRowStyle.chatInputRowBorderRadius,
+                        color: LinagoraSysColors.material().onPrimary,
+                        border: Border.all(
+                          color: LinagoraRefColors.material().tertiary,
+                          width: 1,
+                        ),
+                      ),
+                      decoration: BoxDecoration(
+                        borderRadius:
+                            ChatInputRowStyle.chatInputRowBorderRadius,
+                        color: LinagoraSysColors.material().onPrimary,
+                        border: Border.all(
+                          color: LinagoraRefColors.material().tertiary,
+                          width: 1,
                         ),
                       ),
-                    if (controller.matrix!.isMultiAccount &&
-                        controller.matrix!.hasComplexBundles &&
-                        controller.matrix!.currentBundle!.length > 1)
-                      Container(
-                        height: ChatInputRowStyle.chatInputRowHeight,
-                        alignment: Alignment.center,
-                        child: ChatAccountPicker(controller),
+                      microphoneRequestPermission:
+                          controller.onLongPressAudioRecordInMobile,
+                      startRecording: () {
+                        Logs().d('ChatInputRowMobile:: startRecording');
+                        controller.startRecording.call();
+                      },
+                      stopRecording: (_) {
+                        Logs().d('ChatInputRowMobile:: stopRecording');
+                        if (controller.sendController.text.isNotEmpty) {
+                          controller.sendController.clear();
+                        }
+                        controller.stopRecording.call();
+                      },
+                      sendRequestFunction: (soundFile, time, waveFrom) {
+                        Logs().d(
+                          'ChatInputRowMobile:: sendRequestFunction $soundFile',
+                        );
+                        controller.stopRecording.call();
+
+                        final file = MatrixAudioFile(
+                          bytes: soundFile.readAsBytesSync(),
```

Comment:
When I read that, I'm a bit afraid about the memory usage for long audio because we'll store all the binary in memory right? Is there a way to make a stream or something?